lock.go

```diff
@@ -114,29 +134,35 @@ func (l *Locker) create() (bool, error) {
 			return true, nil
 		}
 
-		if time.Now().Add(l.opts.WaitRetry).After(stop) {
+		if time.Now().Add(l.opts.WaitRetry).After(deadline) {
 			break
 		}
 
-		if l.opts.RetriesCount > 0 && retries <= 0 {
-			break
+		if retryDelay == nil {
+			retryDelay = time.NewTimer(l.opts.WaitRetry)
+			defer retryDelay.Stop()
```

Comment:
bad practice to use `defer` in `for {}`

Reply:
micro-optimisation: I don't want to allocate the timer if there is no need for it